OpenRA.Game/Selection.cs

```diff
@@ -57,9 +57,13 @@ public void Combine(World world, IEnumerable<Actor> newSelection, bool isCombine
 				}
 			}
 
-			var voicedActor = actors.FirstOrDefault(a => a.Owner == world.LocalPlayer && a.IsInWorld && a.HasTrait<IVoiced>());
-			if (voicedActor != null)
-				voicedActor.PlayVoice("Select");
+			var voicedActors = actors.Where(a => a.Owner == world.LocalPlayer && a.IsInWorld && a.HasTrait<IVoiced>());
+			var selectableActors = voicedActors.ToDictionary(a => a, v => v.TraitOrDefault<Selectable>());
+			foreach (var selectable in selectableActors.Where(v => v.Value != null))
+			{
+				selectable.Key.PlayVoice(selectable.Value.Info.Voice);
+				break;
+			}
```

Comment:
+1 to that.
But change `a in actors` to `a in newSelection` and remove the first `TODO`